maven_iuvs/files.py

```diff
@@ -0,0 +1,444 @@
+# Built-in imports
+import fnmatch as fnm
+import glob
+import os
+import warnings
+
+# 3rd-party imports
+from astropy.io import fits
+import numpy as np
+
+
+class Files:
+    """ A Files object stores the absolute paths to files, along with some file handling routines."""
+    def __init__(self, path, patterns):
+        """
+        Parameters
+        ----------
+        path: str
+            The location where to start looking for files.
+        patterns: list
+            List of strings of regex patterns to match filenames to.
+
+        Properties
+        ----------
+        absolute_file_paths: list
+            Absolute file paths of all files matching patterns in path.
+        filenames: list
+            Filenames of all files matching patterns in path.
+
+        Notes
+        -----
+        This class used glob-style matching, so a pattern of ['**/*.pdf'] will recursively search for .pdf files
+        starting from path.
+        """
+        self.__check_path_exists(path)
+        self.__absolute_paths = self.__get_absolute_file_paths_matching_patterns_in_path(path, patterns)
+        self.__filenames = self.__get_filenames_from_absolute_paths(self.__absolute_paths)
+        self.__raise_value_error_if_no_files_found(self.__absolute_paths)
+
+    @property
+    def absolute_file_paths(self):
+        """ Get absolute_file_paths """
+        return self.__absolute_paths
+
+    @property
+    def filenames(self):
+        """ Get filenames """
```

Comment:
I found writing these felt awkward since I had documented those things in the constructor's docstring... which is usually a sign I'm doing something wrong. I moved info on all properties into a more verbose docstring associated with each property, which will be in the next commit. I'll mark this as resolved for now but open to suggestions on my improved way of documenting these.